server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java

```diff
@@ -842,6 +842,9 @@ private void validateComponentName(String componentName) {
     }
 
     private DetailedDeploymentStatus detailedDeploymentStatusFromString(String detailedDeploymentStatus) {
+        if(detailedDeploymentStatus == null || detailedDeploymentStatus.trim().isEmpty()) {
+            return null;
+        }
```

Comment:
Is null really what you want to return? Does the caller of this method handle nulls?

Reply:
The caller of this method only cares about `detailedDeploymentStatus` if the deployment fails and we are sure that this field will populated in this case. My understanding is that the `detailedDeploymentStatus` is not populated only when a deployment transitions to an IN_PROGRESS state and publishes this update. I can take a deeper look at this.
ref: https://github.com/aws-greengrass/aws-greengrass-cli/blob/main/client/src/main/java/com/aws/greengrass/cli/commands/DeploymentCommand.java#L179-L183